deals/module/deals.go

```diff
@@ -235,61 +235,125 @@ func (m *Module) Watch(ctx context.Context, proposals []cid.Cid) (<-chan deals.D
 	return ch, nil
 }
 
-// FinalDealRecords returns a list of all finalized storage deals.
-// Records are sorted ascending by activation epoch then timestamp.
-func (m *Module) FinalDealRecords() ([]deals.DealRecord, error) {
-	ret, err := m.store.getFinalDeals()
-	if err != nil {
-		return nil, fmt.Errorf("getting final deals: %v", err)
+// ListStorageDealRecords lists storage deals according to the provided options.
+func (m *Module) ListStorageDealRecords(opts ...deals.ListDealRecordsOption) ([]deals.StorageDealRecord, error) {
```

Comment:
Here are the new apis to query for deal records. Behavior is specified with the options params.